code/online_challenges/src/project_euler/problem_012/problem_012.cpp

```diff
@@ -0,0 +1,31 @@
+#include<iostream>
+#include<cmath>
+
+int main()
+{
+    int divisorCount = 0;
+    int triangleNumberIndex = 0, triangleNumber = 0;
+    
+    while (divisorCount < 500)
+    { 
+        divisorCount = 0;
+        ++triangleNumberIndex;
+        triangleNumber += triangleNumberIndex ;
+
+        for (int i = 1; i <= pow(triangleNumber, 0.5); ++i)
```

Comment:
A better (and more common condition) is `std::sqrt(triangleNumber) + 1`. (Also fully qualify names)

Reply:
I wonder why this got outdated...